src/main/java/com/checkmarx/flow/service/ProjectNameGenerator.java

```diff
@@ -88,6 +88,8 @@ private String tryGetProjectNameFromScript(ScanRequest request) {
                 Object result = scriptService.runScript(script, bindings);
                 if (result instanceof String) {
                     return ((String) result);
+                } else {
+                    throw new IOException("Script must return a result of type 'java.lang.String'");
                 }
```

Comment:
I think the code should be more tolerant, to keep backward compatibility, and not fail the flow if the return value is not expected.
we probably can add a clear message in the log.warning

@alex-ko-dev what is your opinion on this

Reply:
I would advise throwing an exception, as it is already done for scripted filters here:
com.checkmarx.flow.service.FilterFactory#parseScriptText
com.checkmarx.sdk.service.FilterValidatorImpl#evaluateFilterScript

This will help to avoid "script doesn't work!" tickets from customers.

Reply:
in this case please approve the PR
Thanks